test/functional/eval/executable_spec.lua

```diff
@@ -136,21 +136,62 @@ describe('executable() (Windows)', function()
     eq(1, call('executable', '.\\test_executable_zzz'))
   end)
 
-  it('returns 1 for any existing filename', function()
+  it('returns 1 for any existing filename, when a Unix-shell like \'shell\'', function()
     clear({env={PATHEXT=''}})
+    command('set shell=sh')
     for _,ext in ipairs(exts) do
       eq(1, call('executable', 'test_executable_'..ext..'.'..ext))
     end
     eq(1, call('executable', 'test_executable_zzz.zzz'))
   end)
 
-  it('returns 1 for any existing path (backslashes)', function()
+  it('returns 1 for any existing path, when a Unix-shell like \'shell\' (backslashes)', function()
     clear({env={PATHEXT=''}})
+    command('set shell=bash.exe')
     for _,ext in ipairs(exts) do
       eq(1, call('executable', '.\\test_executable_'..ext..'.'..ext))
       eq(1, call('executable', './test_executable_'..ext..'.'..ext))
     end
     eq(1, call('executable', '.\\test_executable_zzz.zzz'))
     eq(1, call('executable', './test_executable_zzz.zzz'))
   end)
+
+  it('returns 1 for any existing filename, when $PATHEXT contain dot itself', function()
+    clear({env={PATHEXT='.;.zzz'}})
+    command('set shell=sh')
+    for _,ext in ipairs(exts) do
+      eq(1, call('executable', 'test_executable_'..ext..'.'..ext))
+    end
+    eq(1, call('executable', 'test_executable_zzz.zzz'))
+    clear({env={PATHEXT='.zzz;.'}})
```

Comment:
Is this a repeat of the first test? Both set the shell to `sh`.

Reply:
It is strange to do set shell, so it was fixed. If we put it in a loop, I don't like to do that because I don't know which pattern failed.